src/main/java/seedu/duke/task/Item.java

```diff
@@ -0,0 +1,184 @@
+package seedu.duke.task;
+
+import seedu.duke.DukeException;
+import seedu.duke.common.Messages;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+
+// Renamed from previous Task.java with some modifications.
+/**
+ * Represents an item in the list.
+ */
+public class Item implements Comparable<Item> {
+    public static DateTimeFormatter DATETIME_PARSE_FORMAT = DateTimeFormatter.ofPattern("dd-MM-yyyy");
+    public static DateTimeFormatter DATETIME_PRINT_FORMAT = DateTimeFormatter.ofPattern("dd MMM yyyy");
+
+    protected String description;
+    protected boolean isDone;
+    protected int priority;
+    protected String category;
+    protected LocalDate date;
+
+    /**
+     * Constructor used when adding a new task.
+     * By default, the deadline task is not done.
+     *
+     * @param description the description of the task
+     */
+    public Item(String description) {
+        this.description = description;
+        this.isDone = false;
+        this.setPriority(0);
+    }
+
+    /**
+     * Convenience constructor used when loading from the storage file.
+     *
+     * @param description the description of the task
+     * @param isDone true if the task is done already, false otherwise
+     * @param priority the priority of the task
+     */
+    public Item(String description, boolean isDone, int priority) {
+        this.description = description;
+        this.isDone = isDone;
+        this.priority = priority;
+    }
+
+    /**
+     * Retrieves the description of a task.
+     *
+     * @return the description string of the task
+     */
+    public String getDescription() {
+        return description;
+    }
+
+    /**
+     * Retrieves whether the task in done.
+     *
+     * @return true if the task is done already, false otherwise
+     */
+    public boolean getIsDone() {
+        return isDone;
+    }
+
+    /**
+     * Marks the task as done.
+     */
+    public void markAsDone() {
+        isDone = true;
+    }
+
+    /**
+     * Converts the attributes of the task into a formatted string to be saved into the storage file.
+     *
+     * @return the formatted string to be saved into the storage file
+     */
+    public String toFile() {
+        String isDoneString = (isDone) ? "1" : "0";
+        String categoryString = (category == null) ? "" : category;
+        String dateString = getDateString(Item.DATETIME_PARSE_FORMAT);
+        return "T | " + isDoneString + " | " + description + " | " + priority + " | " + categoryString + " | "
+                + dateString;
+    }
```

Comment:
Similarly not every item is going to save in the same format so maybe it might be better to override this in each subclass instead of putting them in the `Item` superclass.

Reply:
Maybe having the fields in `Item` is okay, but it's to put the getters/setters in Tasks or related subclasses so not every `Item` has access to it?

Reply:
Noted. We will work on this later.